.appveyor.yml

```diff
@@ -0,0 +1,198 @@
+install:
+    - ps: |
+        # Check if installation is cached
+        if (!(Test-Path c:\tools\php)) {
+          appveyor-retry cinst --ignore-checksums -y php --version ((choco search php --exact --all-versions -r | select-string -pattern $Env:php | Select-Object -First 1) -replace '[php|]','')
+          # install sqlite
+          appveyor-retry cinst -y sqlite
+
+          cd c:\tools\php
+
+          # Set PHP environment items that are always needed        
+          copy php.ini-production php.ini
+          Add-Content php.ini "`n date.timezone=UTC"
+          Add-Content php.ini "`n extension_dir=ext"
+          Add-Content php.ini "`n extension=php_openssl.dll"
+          Add-Content php.ini "`n extension=php_mbstring.dll"
+          Add-Content php.ini "`n extension=php_fileinfo.dll"
+          Add-Content php.ini "`n extension=php_pdo_sqlite.dll"
+          Add-Content php.ini "`n extension=php_sqlite3.dll"
+
+          # If needed get the MSSQL DLL's
+          if ($env:db -eq "mssql") {
+            $DLLVersion = "4.1.6.1"
+            cd c:\tools\php\ext
+            appveyor-retry appveyor DownloadFile http://windows.php.net/downloads/pecl/releases/sqlsrv/$($:DLLVersion)/php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip
+            7z x -y php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip > $null
+            appveyor-retry appveyor DownloadFile http://windows.php.net/downloads/pecl/releases/pdo_sqlsrv/$($DLLVersion)/php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip
+            7z x -y php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip > $null
+            Remove-Item c:\tools\php\* -include .zip
+            cd c:\tools\php
+            Add-Content php.ini "`nextension=php_sqlsrv.dll"
+            Add-Content php.ini "`nextension=php_pdo_sqlsrv.dll"
+            Add-Content php.ini "`n"
+          }
+
+          # If needed set the the MySQL DLL's
+          if ($env:db -eq "mysql" -Or $env:db -eq "mariadb" ) {
+          Add-Content php.ini "`n extension=php_pdo_mysql.dll"
+          Add-Content php.ini "`n extension=php_mysqli.dll"}
+
+          # If needed set the the postgresql/PGSQL DLL's
+          if ($env:db -eq "postgresql") {Add-Content php.ini "`n extension=php_pdo_pgsql.dll"}
+
+          cd c:\projects\dbal
+
+          if (!(Test-Path c:\projects\dbal\composer.phar)) {
+            appveyor-retry appveyor DownloadFile https://getcomposer.org/composer.phar
+          }
+        }
+    # if needed install mariadb
+    - ps: if ($env:db -eq "mariadb") { appveyor-retry cinst -y mariadb }
+    # install composer dependencies
+    - appveyor-retry php composer.phar self-update
+    - appveyor-retry php composer.phar install --no-progress --profile
+
+before_test:
+# Selectively start the services
+  - ps: >-
+      if ($env:db -eq "mssql") {
+        $instanceName = $env:db_version.ToUpper()
+        Start-Service "MSSQL`$$instanceName"
+      }
+      elseif ($env:db -eq "mysql") {
```

Comment:
looks like the service is not in existence when restoring from cache, not sure how to deal with that since the service is created as part of the MariaDB install. Maybe we will have to install MariaDB for every build.